sos/utilities.py

```diff
@@ -152,7 +155,10 @@ def _child_prep_fn():
                   stderr=STDOUT if stderr else PIPE,
                   bufsize=-1, env=cmd_env, close_fds=True,
                   preexec_fn=_child_prep_fn)
-        stdout, stderr = p.communicate()
+
+        reader = AsyncReader(p.stdout, size_limit)
+        stdout = reader.get_contents()
+
```

Comment:
I suggest adding here `p.poll()`. At least this will properly set `returncode` used later on (and also this is the reason why 2 automated tests fail).

Anyway, I am not an expert on python (sub)processes.

Reply:
Yep - you're right. I've added `p.poll()`